lib/pages/chat/chat_invitation_body.dart

```diff
@@ -0,0 +1,282 @@
+                  ),
+                  child: Text.rich(
+                    textAlign: TextAlign.center,
+                    TextSpan(
+                      children: [
+                        TextSpan(
+                          text: '${_getDisplayInviterName()} \n',
+                          style: Theme.of(context)
+                              .textTheme
+                              .bodyMedium
+                              ?.copyWith(
+                                color: Theme.of(context).colorScheme.primary,
+                              ),
+                        ),
+                        TextSpan(
+                          text: L10n.of(context)!.hasInvitedYouToAChat,
+                          style:
+                              Theme.of(context).textTheme.bodyMedium?.copyWith(
+                                    color: Theme.of(context)
+                                        .colorScheme
+                                        .onSurfaceVariant,
+                                  ),
+                        ),
+                      ],
+                    ),
+                  ),
+                ),
+                Row(
+                  children: [
+                    Expanded(
+                      child: InvitationRejectButton(
+                        onReject: () => onReject(context),
+                      ),
+                    ),
+                    const SizedBox(
+                      width: ChatInvitationBodyStyle.dialogButtonSpacing,
+                    ),
+                    Expanded(
+                      child: InvitationAcceptButton(
+                        onAccept: () => onAccept(),
+                      ),
+                    ),
+                  ],
+                ),
+              ],
+            ),
+          ),
+        ),
+      ),
+    );
+  }
+
+  void onAccept() async {
+    await TwakeDialog.showFutureLoadingDialogFullScreen(
+      future: () async {
+        final waitForRoom = controller.room?.client.waitForRoomInSync(
+          controller.room!.id,
+          join: true,
+        );
+        await controller.room?.join();
+        await waitForRoom;
+      },
+    );
+  }
+
+  void onReject(BuildContext context) async {
+    final result = await showDialog<DialogRejectInviteResult>(
+      context: context,
+      useRootNavigator: false,
+      builder: (c) => const DialogRejectInviteWidget(),
+    );
+
+    if (result == null) return;
+
+    switch (result) {
+      case DialogRejectInviteResult.cancel:
+        return;
+      case DialogRejectInviteResult.reject:
+        return controller.leaveChat();
+    }
+  }
+
+  String _getDisplayInviterName() {
+    if (controller.room!.isDirectChat) {
+      return controller.room!.getLocalizedDisplayname();
+    } else {
+      return controller
+              .room!.lastEvent?.senderFromMemoryOrFallback.displayName ??
+          controller.room!.getLocalizedDisplayname();
+    }
+  }

```

Comment:
IMO: Why don't you write inside ChatController?

Reply:
Done it :)